### ISSUE TITLE
Add custom headers to spec

### ISSUE THREAD

**Issue by janbinx** [WebBreacher/WhatsMyName#748]
As per https://github.com/WebBreacher/WhatsMyName/pull/724#discussion_r1297898147 the usage of the JSON-file for some checkers might require custom headers for the checker to behave as expected.

This is currently a not documented requirement. I suggest adding a `headers` field to the specs. This should probably NOT contain `user-agent` as this will change and is error prone, but all other required headers for both the exist and missing checks to pass.

It could be implementet only for those who need it. This will not break backwards compability and prevent bloating of the JSON.

For Github this could look like so:

```json
{
        "name" : "GitHub",
        "uri_check" : "https://github.com/{account}",
        "e_code" : 200,
        "e_string" : "p-nickname vcard-username d-block",
        "m_string" : "<title>Page not found ",
        "m_code" : 404,
        "known" : ["test", "WebBreacher"],
        "cat" : "coding",
        "headers": {
                "accept": "text/html"
        }
}
```
As @WebBreacher mentions the http2 pseudo-header `authority` in his example a suggestion in the future might be specifying  if a specific version of HTTP is required for the given checker.

Github however does not require HTTP2.

**Comment by WebBreacher** [comment]
And interesting and quite useful addition. I like it. An optional `headers` parameter for those sites requiring it. This would give the downstream checkers more information about how to make a successful query....or they can ignore it altogether.

Anyone else have thoughts about this?

**Comment by janbinx** [comment]
Sorry for the late reply. Well, Insta would be one. It uses a `x-ig-app`-ish on an internal API to query username w/o login

**Comment by WebBreacher** [comment]
I see no downside to this and have altered the schema and submitted it to the project. @janbinx  can you adjust the submitted PR to add this header?

**Comment by janbinx** [comment]
It seems you've set the type for the `headers` to array, not object. This means that if one wants to add key/value pairs it needs to be done like this:

```
headers: [
    {"accept": "application/json"},
    {"x-request-for": "you get the idea"}
]
```

Wouldn't it make more sense to cast it as an object and do this instead:
```
headers: {
    "accept": "application/json",
    "x-request-for": "you get the idea"
}
```
That would certainly be the way you'd use it in a script. 

I've updated the PR for Github as we discussed previously (I totally missing this, sorry), but currently the PR fails the JSON-checkers as the specs expects an array.
